semgrep/semgrep/ignores.py

```diff
@@ -1,14 +1,16 @@
 import fnmatch
 import os
 import re
+from functools import lru_cache
```

Comment:
Need info: We should double-check that these caching changes don't break the backend runner service (maybe we just need to make the caching configurable?)

Reply:
Good point. Many of them use the `self` instance as a cache key btw so that should make them isolated per scan. But we should test this.

Reply:
@nbrahms how can I test this? 🤔